processing/src/main/java/io/druid/segment/filter/BoundFilter.java

```diff
@@ -59,6 +61,11 @@ public boolean apply(String input)
                   if (input == null) {
                     return false;
                   }
+
+                  if (extractionFn != null) {
```

Comment:
This way the predicate supplied to DimensionPredicateFilter is something that operates on extracted values (which seems to make more sense to me)